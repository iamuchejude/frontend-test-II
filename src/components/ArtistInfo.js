import React, { Component, Fragment } from 'react';

class ArtistInfo extends Component {
  constructor(props) {
    super(props);
    this.state = {
      artist: null,
    }
  }

  componentDidMount() {
    const { token } = localStorage;

    fetch(`https://api.spotify.com/v1/artists/1vCWHaC5f2uS3yhpwWbIA6`, {
      headers: {
        'Authorization': `Bearer ${token}`
      }
    }).then(res => res.json())
    .then(data => {
      this.setState({ artist: data })
    })
    .catch(err => console.log(err));
  }

  render() {
    return (
      <Fragment>
        { 
          !this.state.artist ? 'loading.....' :
          <div className="info">
            <div className="photo">
             <img src={ this.state.images[0].url } alt={ this.state.name } />
            </div>
            <div className="profile">
              <ul>
                <li className="name"> { this.state.name } </li>
                <li className="followers"><i className="fa fa-user" /> { this.state.followers.total } Followers </li>
                <li className="genres">
                  {
                    this.state.genres.map(genre => <span>{ genre }</span>)
                  }
                </li>
              </ul>
            </div>
          </div>
        }
    </Fragment>
  )
  }
}

export default ArtistInfo;
