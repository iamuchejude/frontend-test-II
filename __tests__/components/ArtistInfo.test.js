import React from 'react';
import { shallow } from 'enzyme';
import ArtistInfo from '../../src/components/ArtistInfo';

describe('ArtistInfo components', () => {
  test('should render Artist info', () => {
    const wrapper  = shallow(<ArtistInfo />);
    expect(wrapper).toMatchSnapshot();
  });
});
