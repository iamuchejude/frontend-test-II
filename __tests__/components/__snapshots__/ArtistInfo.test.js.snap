// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistInfo components should render Artist info 1`] = `
<Fragment>
  loading.....
</Fragment>
`;
